polynote-frontend/polynote/ui/component/kernel_ui.ts

```diff
@@ -8,55 +8,66 @@ import {TaskInfo, TaskStatus} from "../../data/messages";
 import {errorDisplay} from "./cell";
 import {storage} from "../util/storage";
 import {SocketSession} from "../../comms";
+import {NotebookUI} from "./notebook";
+import {ResultValue} from "../../data/result";
 
 export class KernelUI extends UIMessageTarget {
     private info: KernelInfoUI;
     readonly tasks: KernelTasksUI;
-    readonly symbols: KernelSymbolsUI;
+    readonly symbols?: KernelSymbolsUI;
     readonly el: TagElement<"div">;
     private statusEl: TagElement<"h2">;
     private status: TagElement<"span">;
+    private socket: SocketSession;
+    private path: string;
 
     // TODO: instead of passing path in, can it be enriched by a parent?
-    constructor(eventParent: UIMessageTarget, readonly path: string, showInfo = true, showSymbols = true, showTasks = true, showStatus = true) {
-        super(eventParent);
+    constructor(private notebook?: NotebookUI) {
+        super(notebook);
         this.info = new KernelInfoUI();
-        this.symbols = new KernelSymbolsUI(path).setParent(this);
+        if (notebook) {
+            this.symbols = new KernelSymbolsUI(notebook).setParent(this);
+            this.path = notebook.path;
+            this.socket = notebook.socket;
+        } else {
+            this.socket = SocketSession.global; // TODO: the welcome screen uses a KernelUI to be able to show errors, make a dedicated thing for that and remove this (make notebook non-optional)
```

Comment:
+1, in the future it might be useful to be able to (1) have one place to see errors / logs across notebooks (2) show only system-specific errors/logs